### ISSUE TITLE
Unique index is generating for existing table with OneToOne relationship and primary column

### ISSUE THREAD

**Issue by IvanYur4enk0** [typeorm/typeorm#8485]
### Expected Behavior

1. First Test1.sql is generating with primary key and without unique index. Because unique index is not needed for primary key column.
2. Second Test2.sql is do not generating changes by companyId that is not needed, because companyId is primary key.

### Actual Behavior

1. First Test1.sql is generating primary key without unique index.
2. Second Test2.sql  is generating unique index by companyId column that is not needed, because companyId is primary key.

### Steps to Reproduce

Database: PostgreSQL

1. Create Company and CompanyReport tables with entities.

```typescript
@Entity()
export class Company {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @OneToOne(() => CompanyReport, (report) => report.company)
  report: CompanyReport | null;
}

@Entity()
export class CompanyReport {
  @PrimaryColumn()
  companyId: string;

  @OneToOne(
    () => Company,
    (company) => company.report,
    { nullable: false, onDelete: 'CASCADE' },
  )
  @JoinColumn({
    name: 'companyId',
  })
  company: Company;
}
```

2. Run typeorm migration:generate -c default -n Test1.
3. Run typeorm migration:generate -c default -n Test2.

### My Environment

| Dependency          | Version  |
| Database    |  PostgreSQL 12.9       |
| Node.js version     | 14.16.0  |
| Typescript version  | 4.4.3  |
| TypeORM version     | 0.2.41  |

### Relevant Database Driver(s)

| DB Type              | Reproducible  |
| ---                  | ---           |
| `postgres`           | yes            |

**Comment by codestacx** [comment]
@IvanYur4enk0 Facing same issue. Did you find any solution for this ?

**Comment by KiwiKilian** [comment]
Also experiencing the same. This was reported on #9229 but the example used the decorators wrong. #9282 tried to approach this. The usage combination `PrimaryColumn` and `JoinColumn` [was confirmed to be valid like this](https://github.com/typeorm/typeorm/issues/9229#issuecomment-1227231469):

```typescript
import { Column, Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';
@Entity('testEntity')
export class TestEntity {
    @PrimaryColumn()
    public userId: string;

    @OneToOne('User', 'test')
    @JoinColumn({ name: 'userId' })
    public user: User;

    @Column({ type: 'jsonb', nullable: false })
    public x: string[];
}
```

I will try to setup a testcase.

**Comment by theanurin** [comment]
+1

on typeorm@^0.2.45...

**Comment by maheshdama13** [comment]
+1 on 0.3.11

**Comment by alumni** [comment]
@maheshdama13 Should be fixed for MySQL and others in #9940.
